Remotion/Web/Core/UI/Controls/LazyValidationControl.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Web.UI;
+using System.Web.UI.HtmlControls;
+using Remotion.Utilities;
+
+namespace Remotion.Web.UI.Controls
+{
+  public class LazyValidationControl : HtmlGenericControl
+  {
+    private readonly IValidator? _validator;
+
+    /// <summary>Initializes a new instance of the <see cref="LazyValidationControl" /> class using default values.</summary>
+    protected LazyValidationControl ()
+        : this("span", null)
+    {
+    }
+
+    /// <summary>Initializes a new instance of the <see cref="LazyValidationControl" /> class using the specified tag name.</summary>
+    /// <param name="tag">A string that specifies the tag name of the control. </param>
+    /// <param name="validator">The validator for which the error message should be displayed.</param>
+    public LazyValidationControl (string tag, IValidator? validator)
+        : base(tag)
+    {
+      ArgumentUtility.CheckNotNull(nameof(tag), tag);
+
+      _validator = validator;
+    }
+
+    protected override void Render (HtmlTextWriter writer)
+    {
+      ArgumentUtility.CheckNotNull(nameof(writer), writer);
+
+      if (_validator is null)
+        return;
+
+      if (_validator is ILazyValidator lazyValidator)
+      {
+        lazyValidator.LazyValidate();
+      }
+
+      if (_validator.IsValid)
+        return;
```

Comment:
Depending on the result of the comment about evaluation vs. computing the error message we would need to switch the if statements here.